v2/snmp/manager.go

```diff
@@ -0,0 +1,60 @@
+package snmp
+
+import (
+	"context"
+	"net"
+)
+
+// Manager provides an interface for SNMP device management.
+type Manager interface {
```

Comment:
Is there an naming issue here that we need to worry about.  In SNMP a Manager is defined entity.  Is the risk that someone asks why an SNMP Agent interface is called Manager?

Reply:
Any reason for not using the name Session?

Reply:
Not entirely happy with Manager, but failed to come up with an alternative. 
Did not want to call it a Session (a la netconf) because it's connection-less.  
Could go for Client, I suppose, but it did not feel right in an SNMP context.   
Open to suggestions.